core/virtual_machine_build.go

```diff
@@ -38,24 +44,24 @@ const (
 )
 
 type VirtualMachineBuildArguments struct {
-	Zone                ZoneRef
-	DataCenter          DataCenterRef
+	Zone                *ZoneRef
+	DataCenter          *DataCenterRef
 	Package             VirtualMachinePackageRef
-	DiskTemplate        DiskTemplateRef
+	DiskTemplate        *DiskTemplateRef
 	DiskTemplateOptions []*DiskTemplateOption
-	Network             NetworkRef
+	Network             *NetworkRef
 	Hostname            string
 }
 
 type virtualMachineBuildCreateRequest struct {
 	Hostname            string                   `json:"hostname,omitempty"`
 	Organization        OrganizationRef          `json:"organization"`
-	Zone                ZoneRef                  `json:"zone,omitempty"`
-	DataCenter          DataCenterRef            `json:"data_center"`
-	Package             VirtualMachinePackageRef `json:"package,omitempty"`
-	DiskTemplate        DiskTemplateRef          `json:"disk_template,omitempty"`
+	Zone                *ZoneRef                 `json:"zone,omitempty"`
+	DataCenter          *DataCenterRef           `json:"data_center"`
```

Comment:
I think a `omitempty` flag is missing here :)

```suggestion
	DataCenter          *DataCenterRef           `json:"data_center,omitempty"`
```